assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceTilesWidget/AudienceTiles.js

```diff
@@ -167,6 +194,32 @@ export default function AudienceTiles( { Widget, widgetLoading } ) {
 		)
 	);
 
+	const topCitiesReportError = useSelect( ( select ) => {
+		const citiesError = configuredAudiences.reduce(
+			( acc, audienceResourceName ) => {
+				const error = select( MODULES_ANALYTICS_4 ).getErrorForSelector(
+					'getReport',
+					[
+						{
+							...topCitiesReportOptions,
+							dimensionFilters: { audienceResourceName },
+						},
+					]
+				);
+
+				if ( error ) {
+					individualTileErrors[ audienceResourceName ].push( error );
```

Comment:
The `reduce()` callback should be a pure function, so shouldn't be mutating this external variable.

I've got a proposed refactor, but it's easier to show via a commit which I will push now. I've also realised we don't need to use component state for `allTilesError`, this can also be determined at render time and we can avoid the extra re-render.

Please take a look at my [commit](65331cc669f409e26402f67cb37708f42fa57922), and see what you think.